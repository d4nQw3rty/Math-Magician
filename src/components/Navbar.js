import React from 'react';
import { NavLink } from 'react-router-dom';

const links = [
  {
    id: 1,
    path: '/Home',
    text: 'Home',
  },
  {
    id: 2,
    path: '/calculator',
    text: 'Calculator',
  },
  {
    id: 3,
    path: '/Quote',
    text: 'Quote',
  },
];

const Navbar = () => (
  <header>
    <h1>Math Magicians</h1>
    <nav>
      <ul className="list">
        {links.map((link) => (
          <li key={link.id}>
            <NavLink to={link.path} activeClassName="active-link" className="link" exact>
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  </header>
);

export default Navbar;
