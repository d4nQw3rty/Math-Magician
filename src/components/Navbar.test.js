import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from './Navbar';

describe('testing Navbar component', () => {
  test('Navbar renders correctly', () => {
    const tree = renderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
