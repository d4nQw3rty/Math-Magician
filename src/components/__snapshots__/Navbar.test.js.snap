// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Navbar component Navbar renders correctly 1`] = `
<header>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      className="list"
    >
      <li>
        <a
          activeClassName="active-link"
          className="link"
          exact={true}
          href="/Home"
          onClick={[Function]}
        >
          Home
        </a>
        <span>
          |
        </span>
      </li>
      <li>
        <a
          activeClassName="active-link"
          className="link"
          exact={true}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <span>
          |
        </span>
      </li>
      <li>
        <a
          activeClassName="active-link"
          className="link"
          exact={true}
          href="/Quote"
          onClick={[Function]}
        >
          Quote
        </a>
        <span />
      </li>
    </ul>
  </nav>
</header>
`;
